File: kotlin/app/src/main/kotlin/com/hyperbabel/demo/api/PresenceService.kt
/*
 * HyperBabel API — Presence service.
 */
package com.hyperbabel.demo.api

import com.hyperbabel.demo.data.PresenceHeartbeat
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface PresenceService {
    @POST("api/v1/presence/heartbeat")
    suspend fun heartbeat(@Body body: PresenceHeartbeat)

    @POST("api/v1/presence/status")
    suspend fun status(@Body body: Map<String, String>)

    @GET("api/v1/presence")
    suspend fun list(@Query("user_ids") userIds: String): Map<String, Any>
}

File: swift/HyperBabelDemo/API/PresenceService.swift
/*
 * HyperBabel API — Presence service.
 */
import Foundation

enum PresenceService {
    static func heartbeat(userId: String) async throws -> EmptyResponse {
        try await ApiClient.shared.request(
            "POST", "/presence/heartbeat",
            body: PresenceHeartbeat(userId: userId)
        )
    }

    static func setStatus(userId: String, status: String) async throws -> EmptyResponse {
        try await ApiClient.shared.request(
            "POST", "/presence/status",
            body: ["user_id": userId, "status": status]
        )
    }

    static func list(userIds: [String]) async throws -> [String: String] {
        try await ApiClient.shared.request(
            "GET", "/presence",
            query: [("user_ids", userIds.joined(separator: ","))]
        )
    }
}
